libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/http/HttpDriver.java

```diff
@@ -296,19 +305,11 @@ public String doPost(HttpPost httpPost) throws IOException {
             try (CloseableHttpResponse response = this.client.execute(httpPost)) {
                 if (response != null) {
                     int statusCode = response.getStatusLine().getStatusCode();
-                    switch (statusCode) {
-                        case 200:
-                        case 201:
-                            String out = EntityUtils.toString(response.getEntity());
-                            LOGGER.debug("Data received: {}", out);
-                            return out;
-                        default:
-                            //received bad statuscode, don't bother resending request.
-                            LOGGER.error("Received bad status code: {} from: {} reason: {}", statusCode, url, response.getStatusLine());
-                            //before breaking out. We must close inputstream to prevent from leaking.
-                            response.getEntity().getContent().close();
-                            return "";
-                    }
+                    String out = EntityUtils.toString(response.getEntity());
+                    LOGGER.debug("StatusCode: {} Data received: {}", statusCode, out);
+                    //before breaking out. We must close inputstream to prevent from leaking.
+                    response.getEntity().getContent().close();
```

Comment:
Yes, I agree. I haven't written the original implementation, just wrapped it so I can return HttpDriverResponse and didn't want to make an "out of scope" change that might break the HttpDriver in case I'm missing something.
I went over the implementation again and confirmed that it is closed by a "finally" statement. I'll remove the redundant close call.